.eslintrc.cjs

```diff
@@ -5,12 +5,46 @@ module.exports = {
     'eslint:recommended',
     'plugin:@typescript-eslint/recommended',
     'plugin:react-hooks/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'plugin:prettier/recommended',
```

Comment:
You use the react plugin in plugins: ['react'], but not in extends.
Maybe there is a point to add 'plugin:react/recommended' to the `extends`.
Otherwise it seems that react rules won't be activated.
Check this out